grails-app/migrations/0.9.x/changelog-2024-08-02-1200-add-location-name-index.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9 http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+
+    <changeSet author="kchelstowski" id="020820241200-1">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <indexExists indexName="location_name_idx"/>
+            </not>
+        </preConditions>
+        <createIndex indexName="location_name_idx" tableName="location" unique="false">
```

Comment:
If you look for bin by parent and name, then this should not be required

Reply:
Now that I've read through the entire review I'm understanding this a bit better. And while Artur is right that it probably isn't necessary for this PR, we **should** have an index on location name. However, we need to look into how we're using the name column when querying locations. 

For example, if we always query name and another column together then we'd want to create a composite index. Even if you do some queries with both columns and some queries with just one column, MySQL should use the composite index. So it's better to have the composite index then two separate indexes.

So while there's definitely a need to create an index on location name, we might want to investigate this as a separate issue as long as the performance issue (alluded to below) is no longer an issue.